index.js

```diff
@@ -3,25 +3,33 @@ const { combineReducers, createStore } = require('redux')
 const { mapValues } = require('lodash')
 const { Set } = require('immutable')
 
-const bookReducer = (state = Set(), action) => {
-  if (action.type === 'ADD_BOOK') {
-    return state.add(action.title)
-  }
-  if (action.type === 'REMOVE_BOOK') {
-    return state.delete(action.title)
+function createReducer (handlers) {
+  return function reducer (state = Set(), action) {
+    if (handlers.hasOwnProperty(action.type)) {
+      return handlers[action.type](state, action)
+    } else {
+      return state
+    }
   }
-  return state
 }
 
-const customerReducer = (state = Set(), action) => {
-  if (action.type === 'ADD_CUSTOMER') {
-    return state.add(action.customer)
+const bookReducer = createReducer({
```

Comment:
Just for the fun, let's say that now `books` is no longer a set but a map where the key is the [ISBN](https://en.wikipedia.org/wiki/International_Standard_Book_Number) and the value is a map containing a `title` entry.